### PR TITLE
MMT-3918: Cleanup legacy group permissions when displaying and submitting ACLs

### PR DESCRIPTION
**Pull request by htranho**
# Overview

### What is the feature?

Legacy group permissions causes issues in graphql when submitting update to ACL

### What is the Solution?

Remove legacy group permissions when displaying and submitting ACLs update.

### What areas of the application does this impact?

Editing permissions: Group permission search boxes

# Testing

See test session

### Attachments

See ticket MMT-3918.

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by dmistry1 — COMMENTED** [review]
I see there are two utils where they do the exact same but one checks on `id` and the other checks for `groupId`. You can combine these two by passing the key ('id' or 'groupId') as a parameter into the util.

Also since `items` is an array you can use `.filter()` and avoid `forEach`

```
const removeLegacyPermissions = (groupPermissions, key) => {
  return groupPermissions?.filter(item => !item[key] || validUUID.test(item[key])) || [];
}
```
Then you can call in PermissionForm.jsx using :
`removeLegacyPermissions(permission, 'id')` (For the `id` you are going to have to pull items out of `groups` and pass `items` to the util. 
`removeLegacyPermissions(permission, 'groupId')`

This will reduce code duplication, and using .filter() is more efficient than using .forEach(). With .filter(), we can directly return the resulting array without needing to create and manually populate a new array.

**Review by dmistry1 — APPROVED** [review]

**Review by mandyparson — APPROVED** [review]

File: static/src/js/components/PermissionForm/PermissionForm.jsx [hunk 1]
```diff
@@ -359,9 +359,15 @@ const PermissionForm = ({ selectedCollectionsPageSize }) => {
       const searchAndOrderGroupPermission = []
       const searchPermission = []
 
+      // Removes invalid group permission items, which has no id and no userType
+      // eslint-disable-next-line max-len
+      const removeInvalidGroupPermissionItems = (groupPermissionItems) => groupPermissionItems?.filter((item) => item.id || item.userType)
```
Comment (cgokey) [inline static/src/js/components/PermissionForm/PermissionForm.jsx:364]: Add a comment, stating it returns new array with invalid groups removed.

Comment (cgokey) [inline static/src/js/components/PermissionForm/PermissionForm.jsx:364]: I think you should add a test for this logic as well.

Comment (htranho) [reply static/src/js/components/PermissionForm/PermissionForm.jsx]: Done

Comment (htranho) [reply static/src/js/components/PermissionForm/PermissionForm.jsx]: Added

File: static/src/js/components/PermissionForm/__tests__/PermissionForm.test.jsx [hunk 2]
```diff
@@ -1468,4 +1468,270 @@ describe('PermissionForm', () => {
       })
     })
   })
+
+  describe('form validation', () => {
+    describe('when invalid group permission', () => {
```
Comment (cgokey) [inline static/src/js/components/PermissionForm/__tests__/PermissionForm.test.jsx:1473]: maybe "when invalid group permission occur" 
(reads a little better)

**Review by cgokey — CHANGES_REQUESTED** [review]
Changes look good, I would just modify the test wording slightly.

File: static/src/js/components/PermissionForm/__tests__/PermissionForm.test.jsx (hunk 2 above)
Comment (htranho) [reply static/src/js/components/PermissionForm/__tests__/PermissionForm.test.jsx]: Done

**Pull request merged**